disnake/ext/commands/interaction_bot_base.py

```diff
@@ -698,65 +767,77 @@ async def _sync_application_commands(self) -> None:
-            diff = _app_commands_diff(cmds, current_guild_cmds.values())
-            update_required = bool(diff["upsert"]) or bool(diff["edit"]) or bool(diff["delete"])
-            # Show diff
+        if self._command_sync.sync_global_commands and global_cmds is not None:
+            # Update global commands first
+            diff = _app_commands_diff(
+                global_cmds, self._connection._global_application_commands.values()
+            )
+            if not self._command_sync.allow_command_deletion:
+                # because allow_command_deletion is disabled, we want to never automatically delete a command
+                # so we move the delete commands to delete_ignored
+                diff["delete_ignored"] = diff["delete"].copy()
+                diff["delete"].clear()
+            update_required = bool(diff["upsert"] or diff["edit"] or diff["delete"])
+
+            # Show the difference
             self._log_sync_debug(
                 "Application command synchronization:\n"
-                f"COMMANDS IN {guild_id}\n"
-                "===============================\n"
+                "GLOBAL COMMANDS\n"
+                "===============\n"
                 f"| Update is required: {update_required}\n{_format_diff(diff)}"
             )
-            # Do API requests and cache
+
             if update_required:
+                # Notice that we don't do any API requests if there're no changes.
                 try:
                     to_send = diff["no_changes"] + diff["edit"] + diff["upsert"]
-                    await self.bulk_overwrite_guild_commands(guild_id, to_send)
+                    await self.bulk_overwrite_global_commands(to_send)
                 except Exception as e:
-                    warnings.warn(
-                        f"Failed to overwrite commands in <Guild id={guild_id}> due to {e}",
-                        SyncWarning,
-                    )
+                    warnings.warn(f"Failed to overwrite global commands due to {e}", SyncWarning)
+        # Same process but for each specified guild individually.
+        # Notice that we're not doing this for every single guild for optimisation purposes.
+        # See the note in :meth:`_cache_application_commands` about guild app commands.
+        if self._command_sync.sync_guild_commands and guild_cmds is not None:
+            for guild_id, cmds in guild_cmds.items():
+                current_guild_cmds = self._connection._guild_application_commands.get(guild_id, {})
+                diff = _app_commands_diff(cmds, current_guild_cmds.values())
+                if not self._command_sync.allow_command_deletion:
+                    # because allow_command_deletion is disabled, we want to never automatically delete a command
+                    # so we move the delete commands to delete_ignored
+                    diff["no_changes"] += diff["delete"]
```

Comment:
The key is still `"no_changes"`, should be `"delete_ignored"`

Reply:
https://github.com/DisnakeDev/disnake/pull/806/commits/4fd2b7fe714a807077aec72cce6ff0c599db6d29